infra/event-sync/src/main/java/com/simprints/infra/eventsync/event/commcare/CommCareEventDataSource.kt

```diff
@@ -0,0 +1,164 @@
+                    loadEnrolmentRecordCreationEvents(caseId).collect { emit(it) }
+                }
+            }
+        } catch (e: Exception) {
+            Simber.e("Error while querying CommCare", e)
+            throw e
+        }
+    }
+
+    private fun loadEnrolmentRecordCreationEvents(caseId: String): Flow<EnrolmentRecordCreationEvent> = flow {
+        // Access Case Data Listing for the caseId
+        val caseDataUri = getCaseDataUri().buildUpon().appendPath(caseId).build()
+
+        val cursor = context.contentResolver
+            .query(caseDataUri, null, null, null, null)
+        Simber.d("Cursor for caseId $caseId: $cursor", tag = "CommCareSync")
+        if (cursor != null) {
+            cursor.use { caseDataCursor ->
+                val subjectActions = getSubjectActionsValue(caseDataCursor)
+                Simber.d(subjectActions)
+                val coSyncEnrolmentRecordEvents = parseRecordEvents(subjectActions)
+
+                coSyncEnrolmentRecordEvents
+                    ?.events
+                    ?.filterIsInstance<EnrolmentRecordCreationEvent>()
+                    ?.forEach { emit(it) }
+            }
+        } else {
+            // If listing returned the caseId but the cursor is null, most likely CommCare
+            // logged out in the middle of sync. Throw an exception to retry the worker
+            // instead of thinking sync is complete (and possibly deleting unsynced subjects).
+            throw IllegalStateException("Cursor for caseId $caseId is null")
+        }
+    }
+
+    private fun getSubjectActionsValue(caseDataCursor: Cursor): String {
+        Simber.d("Start looking for subjectActions", tag = "CommCareSync_extra")
+        while (caseDataCursor.moveToNext()) {
+            val key = caseDataCursor.getString(caseDataCursor.getColumnIndexOrThrow(COLUMN_DATUM_ID))
+            if (key == SIMPRINTS_COSYNC_SUBJECT_ACTIONS) {
+                Simber.d("Found subjectActions", tag = "CommCareSync_extra")
+                return caseDataCursor.getString(caseDataCursor.getColumnIndexOrThrow(COLUMN_VALUE))
+            }
+        }
+        Simber.d("No subjectActions found", tag = "CommCareSync_extra")
+        return ""
+    }
+
+    private fun parseRecordEvents(subjectActions: String) = subjectActions.takeIf(String::isNotEmpty)?.let {
+        try {
+            jsonHelper.fromJson<CoSyncEnrolmentRecordEvents>(
+                json = it,
+                module = coSyncSerializationModule,
+                type = object : TypeReference<CoSyncEnrolmentRecordEvents>() {},
+            )
+        } catch (e: Exception) {
+            Simber.e("Error while parsing subjectActions", e)
+            null
+        }
+    }
+
+    private fun getPackageName(): String {
+        return PreferenceManager.getDefaultSharedPreferences(context)
+            .getString(
+                context.getString(com.simprints.infra.resources.R.string.preference_last_calling_package_name_key),
```

Comment:
`import com.simprints.infra.resources.R as IDR`? 

Also, the smallest of nitpicks - could simplify a bunch of methods to expressions instead of blocks.